File: datasets_loader/mnist/load_dataset.py
from torchvision import datasets, transforms

transform = transforms.Compose([transforms.ToTensor(), transforms.Normalize((0.5,), (0.5,))])

train_dataset = datasets.MNIST(root='./dataset', train=True, download=True, transform=transform)
test_dataset = datasets.MNIST(root='./dataset', train=False, download=True, transform=transform)

File: datasets_loader/zinc/load_dataset.go
package zinc

import (
	_ "embed"
	"github.com/Jimmy2099/torch/pkg/fmt"
	"github.com/Jimmy2099/torch/testing"
)

var pythonScript = `
import torch
from torch_geometric.datasets import ZINC
from torch_geometric.loader import DataLoader

# subset: True, False
# split: 'train', 'val', 'test'

train_dataset = ZINC(root='./dataset/ZINC', subset=%s, split='train')
val_dataset = ZINC(root='./dataset/ZINC', subset=%s, split='val')
test_dataset = ZINC(root='./dataset/ZINC', subset=%s, split='test')
`

func loadDataset(subset bool) {
	subSetString := "False"
	if subset == true {
		subSetString = "True"
	}
	testing.RunPyScript(fmt.Sprintf(pythonScript, subSetString, subSetString, subSetString))
}
